load-test-framework/src/main/java/com/google/pubsub/clients/common/LoadTestRunner.java

```diff
@@ -48,21 +48,47 @@
   private static final Stopwatch stopwatch = Stopwatch.createUnstarted();
   private static Server server;
   private static Task client;
-  private static AtomicBoolean finished = new AtomicBoolean(false);
+  private static AtomicBoolean finished = new AtomicBoolean(true);
   private static SettableFuture<Void> finishedFuture = SettableFuture.create();
 
-  public static void run(Function<StartRequest, Task> loadFunction) throws Exception {
-    SettableFuture<StartRequest> requestFuture = SettableFuture.create();
+  private LoadTestRunner(StartRequest request) throws Exception {
+    log.info("Request received, starting up server.");
+    ListeningExecutorService executor = MoreExecutors.listeningDecorator(
+        Executors.newFixedThreadPool(request.getMaxOutstandingRequests() + 10));
+
+    final RateLimiter rateLimiter = RateLimiter.create(request.getRequestRate());
+    final Semaphore outstandingTestLimiter =
+        new Semaphore(request.getMaxOutstandingRequests(), false);
+
+    final long toSleep = request.getStartTime().getSeconds() * 1000 - System.currentTimeMillis();
+    if (toSleep > 0) {
+      Thread.sleep(toSleep);
+    }
+
+    stopwatch.start();
+    while (shouldContinue(request)) {
```

Comment:
This isn't about how many times it will be invoked, this is a cleanup focused around being agnostic to whether it is supposed to run for a duration or for a particular number of messages.